source/dev/coding_guide.rst

```diff
@@ -41,6 +42,27 @@ generating testable code is not just about making testing less troublesome, but
 making the code itself more robust, and easier to maintain.
 
 
+Why is using existing code preferable?
+--------------------------------------
+The open-source ecosystem is a wonderful place, full of useful code that has already been developed.
+In many cases, this code is the result of dozens of people's work over many years, and has already
+gone through rigorous review to ensure that it is accurate, robust, efficient, and well-documented.
+By taking advantage of existing code, developers can focus on improving the quality of the things
+in their code that are more novel, and reviewers can focus their review on this new code, rather than
+both sides having to focus on duplicative code.
+
+For example, in Python:
++ If you want to operate on arrays or matrices of un-labelled data, you probably want to use NumPy (and/or SciPy).
++ If you want to operate on tables of labelled data, you probably want to use Pandas.
++ If you want to analyze a collection of nodes, connected by links, you probably want to use networkx.
+
+It's not a requirement that you work with third-party libraries, but if what you are manually implementing overlaps
+substantially with an existing mature library (and especially if this is a library that we already use within a project),
+we may suggest that the code be re-written before being merged. If you are not familiar with this library, this is okay:
+if you let us know that you are unfamiliar, we may be able to provide a code snippet or a git branch that
+demonstrates this functionality.
+
```

Comment:
Maybe we could also add here if anyone finds a more efficient implementation in our existing code, it is very welcomed to point it out or submit a PR to refactor and we will appreciate the contribution.

Reply:
Perhaps we should reorganize the contribution guide. We could have a first section that explicitly lists the type of contribution with short examples: bug report, feature request, refactoring, documentation, testing, etc. Then, we can go in the details of git, docstring, code guide, etc. 

I was thinking about doing something similar for the installation guide. Depending on what the user wants to do, specify the type of installation needed.